aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -52,10 +51,77 @@ Tensor _bincount_cpu_template(
   }
   return output;
 }
+
+// A helper function to get the bin for histogram
+// while each bin is inclusive at the lower end and exclusive at the higher,
+// i.e. [start, end) the last bin is inclusive at both, i.e. [start, end], in
+// order to include maxvalue if exists
+template <typename input_t>
+inline int64_t getbin(input_t x, input_t min, input_t max, int64_t bins) {
+  if (x == max)
+    return bins - 1;
+  return (int64_t)((x - min) * bins / (max - min));
+}
+
+///////////////// histogram /////////////////
+template <typename input_t, typename weights_t>
+Tensor _histogram_cpu_template_uniformbins(
+    const Tensor& self,
+    int64_t nbins,
+    const Tensor& weights,
+    input_t min,
+    input_t max,
+    bool density) {
+
+  if (min == max) {
```

Comment:
Can you elaborate on what's going on here?

The code may need a comment, too.

Reply:
It's to avoid divide by zero when the range consists of only one point. The result is technically incorrect in that case, but the same behavior can be observed in NumPy.
```
>>> np.histogram([0, 0, 0, 0, 0, 1, 1, 1], range=(.5, .5))
(array([5, 0, 0, 0, 0, 0, 0, 0, 0, 3], dtype=int64),
 array([0. , 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1. ]))
```

Reply:
Thank you for explaining that. Should we just throw an error in this case, then? An error message now gives us flexibility changing the behavior later. Is there any utility for the NumPy behavior?

Reply:
Added an error message for explicitly specifying min = max to have flexibility. However, there might be utility for the NumPy behavior - such as when the Array to be histogrammed consists of only one value, possibly repeated multiple times, and the user intends to use a histogram with outlier removal.

If min and max are computed from the input Tensor, this computation is, however, correct, so I'm keeping that there. This is also much more likely to occur than the previous case, as if it was solved the way above, histogramming an array containing only one value (possibly repeated multiple times) would fail for no apparent reason.